### ISSUE TITLE
Remove objects read permission

### ISSUE THREAD

**Issue by ksuderman** [galaxyproject/galaxy-helm#480]
Galaxy 24.1.1 will include the fix that will allow us to remove read permission for the entire objects directory.

**Comment by afgane** [comment]
The [fix](https://github.com/galaxyproject/galaxy/pull/18462) is against release 24.0 actually so we won't have to wait for the next release.

**Comment by ksuderman** [comment]
True, but the branch won't be tagged so we won't get a Docker image built automatically (except for the 24.0-auto image).  According to John D 24.1.1 will be released in a few days so it should be ok to wait until then.

**Comment by DuttaAnik** [comment]
Hi @ksuderman have you released the` 24.1.1` version with the `objects` folder permission bug fixed?

**Comment by ksuderman** [comment]
Hi @Anik, I did not see a notification about the release, but I see galaxy-min:24.1.1 on quay.io.  I am testing it now and will report back.

**Comment by ksuderman** [comment]
Our tests are passing and #481 will be merged shortly.

**Comment by DuttaAnik** [comment]
Alright. This morning I pulled the 24.1.1 that was available and still the permission denied issue persisted in the object folder. So, I was wondering if the issue was resolved or not.

**Comment by ksuderman** [comment]
Is that with Kraken?  I will try to reproduce using that.

**Comment by DuttaAnik** [comment]
Yes, it is with Kraken. I also tried with Diamond database builder and faced the same error of permission denied.

**Comment by DuttaAnik** [comment]
Hi @ksuderman could you reproduce and resolve the error? Please let me know.

**Comment by ksuderman** [comment]
Yes I am able to reproduce, at least I think it is the same problem you are experiencing.  To be sure though, it is the `data_manager_build_kraken2_database` tool that is the problem and not `kraken2` itself?

I think this is a separate, but related, bug.  I am trying to get a few more data points before reaching out to the backend team to find a solution.

However, is there a reason you are trying to build the Kraken2 database manually rather than using the ones available on CVMFS?  I see you have `refdata.enabled=false` in your `values.yaml`, at least the values shared on the GalaxyHelp channel.  It looks like the database you want to use is available as part of the default reference data included on CVMFS.

```
refdata:
  enabled: true
  type: cvmfs
cvmfs:
  deploy: true  # unless you are deploying CVMFS separately from Galaxy
```
<img width="577" alt="Screenshot 2024-07-09 at 12 34 01 PM" src="https://github.com/galaxyproject/galaxy-helm/assets/4098072/ea87d3ac-8b59-4834-8df0-061d68e72611">

**Comment by DuttaAnik** [comment]
Hi @ksuderman thanks for reproducing the error. Yes, the problem is with the kraken2 database builder tool. Kraken2 itself works fine (or would work fine if there is a database). 
Regarding your second question, I need to build the database using the Prebuilt RefSeq indexes which are available and getting updated once or twice a year. Also, there is an option of building a custom database with reference genomes other than NCBI. I am not using that at the moment. I am just trying to build it from the available ones and getting the error. For the time being, I can try to change the value file and use your suggestion.

**Comment by DuttaAnik** [comment]
Hi @ksuderman, I was wondering if there is any update on this issue. Thanks.

**Comment by ksuderman** [comment]
Unfortunately not at this time.  The problem is in the way Galaxy runs the post job event handler, which runs in the Galaxy process (pod) and not by the tool.  We are still trying to determine why that event handler does not seem to have the proper volumes mounted.

**Comment by DuttaAnik** [comment]
Hi @ksuderman, I was wondering if there is any update on this issue. Thanks.

**Comment by ksuderman** [comment]
Sorry for the delayed response, I was on vacation the past few weeks. This is on the top of my to-do list.
